libs/sdk-core/src/lib/db/search/search.ts

```diff
@@ -2,6 +2,45 @@ import { catchError, map, of, tap } from 'rxjs';
 import type { INuclia } from '../../models';
 import type { Search, SearchOptions } from './search.models';
 
+export const find = (
+  nuclia: INuclia,
+  kbid: string,
+  path: string,
+  query: string,
+  features: Search.Features[] = [],
+  options?: SearchOptions,
+  useGet?: boolean,
+) => {
+  const params: { [key: string]: string | string[] } = {};
+  if (options?.isAdvanced) {
+    params['advanced_query'] = query || '';
+  } else {
+    params['query'] = query || '';
+  }
+  params['features'] = features;
+  const { inTitleOnly, ...others } = options || {};
+  if (inTitleOnly) {
+    params['fields'] = ['a/title'];
+  }
+
+  params['shards'] = nuclia.currentShards?.[kbid] || [];
+
+  const searchMethod = useGet
+    ? nuclia.rest.get<Search.FindResults | { detail: string }>(`${path}/find?${serialize(params, others)}`)
```

Comment:
I feel like it would worth having a model `ErrorResponse` for this `{ detail: string }`, because it's actually common for all the endpoints, don't you think?

Reply:
Right, we will do that after this refactoring.